infra/discovery/src/listener/listener.js

```diff
@@ -134,84 +136,92 @@ async function main() {
   do {
     start = new Date()
 
-    // Compute the range of blocks to process,
-    // while respecting trailing block configuration.
+    // Compute the range of blocks to process
     const currentBlock = await withRetrys(async () => {
       return context.web3.eth.getBlockNumber()
     })
-    const toBlock = Math.max(currentBlock - context.config.trailBlocks, 0)
+    const toBlock = Math.max(currentBlock, 0)
 
-    // Listener is up to date. Nothing to do.
-    if (toBlock <= processedToBlock) {
-      logger.debug('No new blocks to process')
-      continue
-    }
-    logger.info(
-      `Querying events within interval (${processedToBlock}, ${toBlock}]`
-    )
+    for (const contractKey of Object.keys(contracts)) {
+      let processedToBlock = await getLastBlock(
```

Comment:
nit: maybe add a debug statement to show which contract is being processed. like
logger.debug(`Processing ${contractKey}`)

Reply:
Same as above